From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Mon, 17 Nov 2025 17:01:52 -0600
Subject: [PATCH] ntdll: HACK: Avoid unitialized Dr0-Dr3 in RtlRaiseException
 on x64.

CW-Bug-Id: #26231
Valve-commit: 5687b3671cdfd44d44416af4ef80d409144aa942
Picked: clean
Class: valve
Evidence: docs/evidence/2026-09-28-wine-proton-rebase.md
Offered-upstream: no
---
 dlls/ntdll/signal_x86_64.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dlls/ntdll/signal_x86_64.c b/dlls/ntdll/signal_x86_64.c
index 0acbd68..44bb01e 100644
--- a/dlls/ntdll/signal_x86_64.c
+++ b/dlls/ntdll/signal_x86_64.c
@@ -840,6 +840,10 @@ __ASM_GLOBAL_FUNC( RtlRaiseException,
                    "movq %rax,0x10(%rcx)\n\t"   /* rec->ExceptionAddress */
                    "xor %rax,%rax\n\t"
                    "movq %rax,0x70(%rdx)\n\t"   /* Context->Dr7 */
+                   "movq %rax,0x48(%rdx)\n\t"   /* Context->Dr0 */
+                   "movq %rax,0x50(%rdx)\n\t"   /* Context->Dr1 */
+                   "movq %rax,0x58(%rdx)\n\t"   /* Context->Dr2 */
+                   "movq %rax,0x60(%rdx)\n\t"   /* Context->Dr3 */
                    "movl $1,%r8d\n\t"
                    "movq %gs:0x60,%rax\n\t"     /* Peb */
                    "cmpb $0,0x02(%rax)\n\t"     /* BeingDebugged */
